manager/apis/app/v1alpha1/m4dapplication_types.go

```diff
@@ -87,7 +75,12 @@ type M4DApplicationSpec struct {
 	// +optional
 	Selector Selector `json:"selector"`
 
-	// AppInfo contains information describing the reasons and geography of the processing
+	// UserSecretRef points to the secret that holds user role and credentials.
```

Comment:
It holds the role and credentials to what?  There are 3 potential external systems to which M4D is connected: catalog, policy manager, credential manager.  To which of these 3 does this refer?  Obviously there are certain cases where there is one system that plays all those roles, but that isn't always the case.

Reply:
These are user credentials to connect to the catalog. Currently, only the catalog requires authentication: policy manager uses the credentials to retrieve information from the catalog and does not require authentication of its own. Credential manager is going to be removed in the future version.
If more credentials are required, we can change the secret structure to include credentials to other systems as well.

Reply:
> does not require authentication of its own

That's a wrong assumption.

Reply:
User credentials to all relevant systems are stored in a single secret where key is the system name, and value is the credentials to this system. For example, secret.StringData["Egeria"] = '{"user": "xxx", "password": "yyy"}'